v4/core/utils_test.go

```diff
@@ -192,3 +194,221 @@ func TestPointers(t *testing.T) {
 	var float64Var = float64(23)
 	assert.Equal(t, &float64Var, Float64Ptr(float64Var))
 }
+
+func TestConvertSliceFloat64(t *testing.T) {
+	float64Slice := []float64{float64(9.56), float64(4.56), float64(2.4)}
+	expected := []string{"9.56", "4.56", "2.4"}
+	convertedSlice := ConvertSlice(float64Slice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.NotEmpty(t, convertedSlice)
+	assert.Equal(t, expected, convertedSlice)
+
+	float64Slice = []float64{}
+	convertedSlice = ConvertSlice(float64Slice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.Empty(t, convertedSlice)
+}
+
+func TestConvertSliceFloat32(t *testing.T) {
+	float32Slice := []float32{float32(9.56), float32(4.56), float32(2.4)}
+	expected := []string{"9.56", "4.56", "2.4"}
+	convertedSlice := ConvertSlice(float32Slice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.NotEmpty(t, convertedSlice)
+	assert.Equal(t, expected, convertedSlice)
+
+	float32Slice = []float32{}
+	convertedSlice = ConvertSlice(float32Slice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.Empty(t, convertedSlice)
+}
+
+func TestConvertSliceInt64(t *testing.T) {
+	int64Slice := []int64{int64(38), int64(26), int64(22)}
+	expected := []string{"38", "26", "22"}
+	convertedSlice := ConvertSlice(int64Slice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.NotEmpty(t, convertedSlice)
+	assert.Equal(t, expected, convertedSlice)
+
+	int64Slice = []int64{}
+	convertedSlice = ConvertSlice(int64Slice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.Empty(t, convertedSlice)
+}
+
+func TestConvertSliceInt(t *testing.T) {
+	intSlice := []int{3, 2, 1}
+	expected := []string{"3", "2", "1"}
+	convertedSlice := ConvertSlice(intSlice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.NotEmpty(t, convertedSlice)
+	assert.Equal(t, expected, convertedSlice)
+
+	intSlice = []int{}
+	convertedSlice = ConvertSlice(intSlice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.Empty(t, convertedSlice)
+}
+
+func TestConvertSliceByteArray(t *testing.T) {
+	testString := "test string 1..."
+	byteArray := []byte(testString)
+	byteArraySlice := [][]byte{byteArray}
```

Comment:
Let's have at least 2 elements in the slice.

Reply:
Done